src/lib/forms-validation/common.ts

```diff
@@ -4,15 +4,29 @@ import Joi from 'joi'
 
 export function validateJoiSchema<T>(
   data: unknown,
-  schema: Joi.Schema,
-  options?: Joi.ValidationOptions,
-): T {
-  const result = schema.validate(data, options)
+  schema: Joi.Schema<T>,
+):
+  | {
+      success: true
+      data: T
+    }
+  | {
+      success: false
+      error: Joi.ValidationError
+    } {
+  const result = schema.validate(data, {
+    stripUnknown: true,
+  })
   if (result.error) {
-    throw result.error
```

Comment:
If we kept throwing this we could have kept the function flows as they were instead of needing to return early all over the place.